### PR TITLE
Cleanups: remove CFeatures::list_feature_obj (#4705)

### PR DESCRIPTION
**Pull request by ktiefe**

### CODE REVIEW

File: src/shogun/features/CombinedDotFeatures.cpp [hunk 1]
```diff
@@ -45,17 +45,17 @@ CCombinedDotFeatures::~CCombinedDotFeatures()
 void CCombinedDotFeatures::list_feature_objs()
```
Comment (karlnapf) [inline src/shogun/features/CombinedDotFeatures.cpp]: this method here naturally can also be removed ...

**Comment by karlnapf** [comment]
Thanks for the PR. I made a comment that you can also remove the method that you refactored... otherwise we can merge this soon

File: src/shogun/features/CombinedDotFeatures.cpp (hunk 1 above)
Comment (ktiefe) [reply src/shogun/features/CombinedDotFeatures.cpp]: done

File: src/shogun/features/CombinedFeatures.cpp [hunk 2]
```diff
@@ -92,9 +77,9 @@ bool CCombinedFeatures::check_feature_obj_compatibility(CCombinedFeatures* comb_
 				SG_UNREF(f1);
 				SG_UNREF(f2);
 				SG_INFO("not compatible, combfeat\n")
-				comb_feat->list_feature_objs();
+				SG_INFO(comb_feat->to_string().c_str())
```
Comment (karlnapf) [inline src/shogun/features/CombinedFeatures.cpp]: I am not sure `SG_INFO` acutally works this way. Wouldn't it have to be
`SG_INFO("%s\n", comb_feat->to_string.c_ctr());`?

Comment (karlnapf) [reply src/shogun/features/CombinedFeatures.cpp]: Did you test this and look at the output?

Comment (ktiefe) [reply src/shogun/features/CombinedFeatures.cpp]: I run just UT, but it doesn't cover that part of code. anyway I run: in 
TEST(CombinedFeaturesTest,test_array_operations) thi spart of code
std::cout<<"\n"<<comb_feat->to_string().c_str()<<"\n";

this is output:

CombinedFeatures(cache_size=0 ,feature_array=DynamicObjectArray(...) ,num_subscriptions={function},num_vec=3 ,preproc=DynamicObjectArray(...) ,properties=0 ,subset_stack=SubsetStack(...) )

Of course I can remove list_feature_obj() methode from the code at all but as I understand I should replace it with to_string().

Comment (ghost) [reply src/shogun/features/CombinedFeatures.cpp]: Hi ktiefe. Please check that the output ends with a newline character. I suspect that if you did `SG_INFO(comb_feat->to_string().c_str()); SG_INFO("Hello, World\n");` the output is going to be concatenated as follows
```
CombinedFeatures(cache_size=0 ,feature_array=DynamicObjectArray(...) ,num_subscriptions={function},num_vec=3 ,preproc=DynamicObjectArray(...) ,properties=0 ,subset_stack=SubsetStack(...) )Hello, World
```
If that's the case, what @karlnapf suggested seems to be the proper approach.

**Comment by karlnapf** [comment]
hi! :)
any updates?

**Comment by karlnapf** [comment]
This looks good to me!
Ill wait for other to comment and then merge

**Comment by ktiefe** [comment]
Thats Great!

wt., 13 sie 2019, 16:23 użytkownik Heiko Strathmann <
notifications@github.com> napisał:

**Comment by ghost** [comment]
@ktiefe Hi! There have been a couple of changes related to how logging is done. Now instead of `SG_INFO` we use `io::info`, and instead of printf style flags, we use python style flags. So
```C++
SG_INFO("%s\n", comb_feat->to_string.c_ctr());
```
would be
```C++
io::info("{}", comb_feat->to_string()); // without new line at the end
```
For this to be mergeable, please rebase to the latest head of the branch, and use the new functions.

**Comment by ktiefe** [comment]
Hi,

Thanks for hint. I will correct it and rebase.

br,
KG

**Comment by ktiefe** [comment]
Hi,
Done
br,
KG

**Comment by vigsterkr** [comment]
@ktiefe great stuff thnx!

**Pull request merged**